### ISSUE TITLE
Deadlock with large files

### ISSUE THREAD

**Issue by giln** [ejeinc/MetalScope#5]
Hello,

We experience some deadlock when using large images (around 50mb).

It seems the issue is with the DeviceOrientationProvider.

Commenting the following lines in the StereoView fixes the deadlock.

```
if let provider = orientationNode.deviceOrientationProvider, provider.shouldWaitDeviceOrientation(atTime: time) {
            provider.waitDeviceOrientation(atTime: targetTime)
        }
```

**Comment by giln** [comment]
It seems the issue is not caused by the large files but because our App is configured only for Landscape Left Orientation.
Since it is locked in only 1 orientation, it waits indefinitely for device Orientation.

**Comment by junpluse** [comment]
Hi @giln,

Thanks for your report, and sorry for the late reply. 🙇 
Is there an example project to demonstrate the issue?

**Comment by junpluse** [comment]
I fixed a potential infinite waiting issue in #7, and released it as a part of [0.6.0](https://github.com/ejeinc/MetalScope/releases/tag/0.6.0).
Please try the latest version in your project.
